figures/dsp/dsp-arq.mmd

```diff
@@ -0,0 +1,15 @@
+sequenceDiagram
+    participant Sender
+    participant Receiver
+
+    Sender->>Receiver: Transmit Packet
+    Note over Sender, Receiver: Packet transmission
+    alt Packet received successfully
+        Receiver->>Sender: ACK
+        Note over Sender, Receiver: Acknowledgment received
+    else Packet lost or corrupted
```

Comment:
How does the receiver know if the packet is lost? They never got it, so they don't know its there.

Reply:
I would assume if the request times out or like you said if there is an absence of the packet.

Reply:
Ah okay I get what you mean